File: rust/lib/src/services/service.rs
use downcast_rs::{Downcast, impl_downcast};

pub type ServiceId = String;

// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
pub trait Service: Downcast {
    // Returns true if the service is available
    fn is_available(&self) -> bool;
}
impl_downcast!(Service);

File: golang/lib/services/service.go
/*
 * Copyright (c) 2020 - present Kurtosis Technologies LLC.
 * All Rights Reserved.
 */

package services

/*
The identifier used for services with the network.
*/
type ServiceID string

// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
type Service interface {
	// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
	IsAvailable() bool
}
